apps/transport/lib/jobs/multi_validation_with_error_notification_job.ex

```diff
@@ -66,55 +96,120 @@ defmodule Transport.Jobs.MultiValidationWithErrorNotificationJob do
+    )
+  end
+
+  defp relevant_static_validations(%DateTime{} = datetime_limit) do
+    validator_names = Enum.map(@static_data_validators, & &1.validator_name())
 
     DB.MultiValidation.base_query()
-    |> where([mv], mv.max_error in ["Error", "Fatal"] or fragment("?->>'has_errors' = 'true'", mv.result))
     |> where(
-      [mv],
+      [multi_validation: mv],
+      mv.max_error in ["Error", "Fatal"] or fragment("?->>'has_errors' = 'true'", mv.result)
+    )
+    |> where(
+      [multi_validation: mv],
       not is_nil(mv.resource_history_id) and mv.validator in ^validator_names and mv.inserted_at >= ^datetime_limit
     )
     |> preload(resource_history: [resource: [:dataset]])
     |> DB.Repo.all()
     |> Enum.group_by(& &1.resource_history.resource.dataset)
   end
 
-  defp subscriptions(%DB.Dataset{} = dataset, role) do
+  defp relevant_realtime_validations(%DateTime{} = datetime_limit) do
+    validator_names = Enum.map(@realtime_data_validators, & &1.validator_name())
+
+    DB.MultiValidation.base_query()
+    |> where([multi_validation: mv], fragment("?->>'has_errors' = 'true'", mv.result))
+    |> where(
+      [multi_validation: mv],
+      is_nil(mv.resource_history_id) and mv.validator in ^validator_names and mv.inserted_at >= ^datetime_limit
+    )
+    |> preload(resource: :dataset)
+    |> DB.Repo.all()
+    |> Enum.group_by(& &1.resource.dataset)
+  end
+
+  defp subscriptions(%DB.Dataset{} = dataset, role, validator_name) do
     @notification_reason
     |> DB.NotificationSubscription.subscriptions_for_reason_dataset_and_role(dataset, role)
-    |> reject_already_sent(dataset)
+    |> reject_already_sent(dataset, validator_name)
   end
 
-  defp reject_already_sent(notification_subscriptions, %DB.Dataset{} = dataset) do
-    already_sent_emails = email_addresses_already_sent(dataset)
+  defp reject_already_sent(notification_subscriptions, %DB.Dataset{} = dataset, validator_name) do
+    already_sent_emails = email_addresses_already_sent(dataset, validator_name)
 
     Enum.reject(notification_subscriptions, fn %DB.NotificationSubscription{contact: %DB.Contact{email: email}} ->
       email in already_sent_emails
     end)
   end
 
-  def email_addresses_already_sent(%DB.Dataset{id: dataset_id}) do
-    datetime_limit = DateTime.utc_now() |> DateTime.add(-@nb_days_before_sending_notification_again, :day)
+  def all_validators, do: @static_data_validators ++ @realtime_data_validators
+
+  @doc """
+  iex> sending_delay_by_validator(Transport.Validators.GBFSValidator.validator_name())
+  30
+  iex> all_validators() |> Enum.map(& &1.validator_name()) |> Enum.each(&sending_delay_by_validator/1)
+  :ok
+  """
+  @spec sending_delay_by_validator(binary()) :: pos_integer()
+  def sending_delay_by_validator(validator) do
+    %{
+      Transport.Validators.GTFSTransport => 7,
+      Transport.Validators.TableSchema => 7,
+      Transport.Validators.EXJSONSchema => 7,
+      Transport.Validators.GBFSValidator => 30
```

Comment:
Que penses-tu de rajouter l'unité ici pour rendre ça plus lisible et aussi plus souple selon les validateurs ?

```suggestion
      Transport.Validators.GTFSTransport => {7, :day},
      Transport.Validators.TableSchema => {7, :day},
      Transport.Validators.EXJSONSchema => {7, :day},
      Transport.Validators.GBFSValidator => {30, :day}
```

Reply:
Oui, j'aimerais avoir une durée mieux typée ! Je vais voir si je peux utiliser https://hexdocs.pm/elixir/Duration.html

Reply:
Malheureusement ça arrive dans Elixir 1.17 et on est en 1.16 https://hexdocs.pm/elixir/changelog.html#adding-duration-and-shift-2-functions.

J'ai fait un truc correct https://github.com/etalab/transport-site/pull/4352/commits/bea508bf4084db60b40f741a1671efe169a5d2c1